datastore/src/main/java/com/ds/datastore/RequestListener.java

```diff
@@ -0,0 +1,26 @@
+package com.ds.datastore;
+
+import java.util.UUID;
+
+import javax.servlet.ServletRequestEvent;
+import javax.servlet.ServletRequestListener;
+import javax.servlet.annotation.WebListener;
+import javax.servlet.http.HttpServletRequest;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@WebListener
+public class RequestListener implements ServletRequestListener{
+    private Logger logger = LoggerFactory.getLogger(RequestListener.class);
+    @Override
+    public void requestInitialized(ServletRequestEvent sre){
+        HttpServletRequest request = (HttpServletRequest)sre.getServletRequest();
+        String requestID = request.getHeader("requestID");
+        if(requestID == null){
+            requestID = String.valueOf(UUID.randomUUID());
+        }
+        request.setAttribute("requestID", requestID);
+        logger.info("Call from {}, Request Type: {}, Request {} received", request.getHeader("referer"), request.getMethod(), requestID);
```

Comment:
1) The referer may not be populated (in the case of the initial request)
2) If I understand correctly, the referer doesnt provide the full URI - only the server address and book store controller

`requst.getRequestURI` I believe will be much more helpful

Reply:
I think it counts as cheating when you have way more knowledge about these methods

Reply:
LOL. AKA "experience"